netbox/netbox_inventory.py

```diff
@@ -132,53 +182,84 @@ def fetch_inventory(cli_opts):
     if cli_opts.region:
         params["region"] = cli_opts.region
 
-    res = netbox.get("/api/dcim/devices/", params=params)
+    res = NETBOX_SESSION.get("/api/dcim/devices/", params=params)
     if not res.ok:
         sys.exit("FAIL: get inventory: " + res.text)
 
     body = res.json()
     device_list = body["results"]
 
-    # -------------------------------------------------------------------------
-    # User Filters
-    # -------------------------------------------------------------------------
+    return apply_filters(device_list, cli_opts)
+
+
+def apply_filters(device_list: list[dict], cli_opts: argparse.Namespace) -> Iterator[dict]:
+    """Apply user-provided filters to the device list.
 
-    # If Caller provided an explicit list of device-roles, then filter the
-    # device list based on those roles before creating the inventory
+    Args:
+        device_list: The list of devices to filter.
+        cli_opts: Parsed command-line options.
 
+    Returns:
+        Iterator[dict]: The filtered list of devices.
+    """
     filter_functions = []
 
     if cli_opts.role:
 
-        def filter_role(dev_dict):
+        def filter_role(dev_dict: dict) -> bool:
+            """Filter to include devices that match the specified roles.
+
+            Args:
+                dev_dict (dict): The device record to filter.
+
+            Returns:
+                bool: True if the device's role matches the specified roles, False otherwise.
+            """
             return dev_dict["device_role"]["slug"] in cli_opts.role
 
         filter_functions.append(filter_role)
 
     if cli_opts.exclude_role:
 
-        def filter_ex_role(dev_dict):
+        def filter_ex_role(dev_dict: dict) -> bool:
+            """Filter to exclude devices that match the specified roles.
+
+            Args:
+                dev_dict (dict): The device record to filter.
+
+            Returns:
+                bool: True if the device's role does not match the excluded roles, False otherwise.
+            """
             return dev_dict["device_role"]["slug"] not in cli_opts.exclude_role
 
```

Comment:
**Suggestion:** Refactor the `apply_filters` function to separate concerns by moving the filter function definitions outside of the main function body. [maintainability, importance: 6]
```suggestion
        def filter_role(dev_dict: dict, roles) -> bool:
            return dev_dict["device_role"]["slug"] in roles
        def filter_ex_role(dev_dict: dict, excluded_roles) -> bool:
            return dev_dict["device_role"]["slug"] not in excluded_roles
        filter_functions.append(lambda dev: filter_role(dev, cli_opts.role))
        filter_functions.append(lambda dev: filter_ex_role(dev, cli_opts.exclude_role))
```